src/components/utils/utils.ts

```diff
@@ -0,0 +1,90 @@
+import html2canvas from "html2canvas";
+
+export const download = async (id: string): Promise<void> => {
+  const element: HTMLElement = document.getElementById(id);
+
+  let canvas: HTMLCanvasElement;
+
+  canvas = await html2canvas(element, {
+    scale: 2, // Increase resolution
+  });
+
+  // Convert canvas to blob
+  const blob = await new Promise<Blob>((resolve) => {
+    canvas.toBlob((blob) => {
+      resolve(blob as Blob);
+    }, "image/jpg");
+  });
+
+  // Create download link
+  const url: string = URL.createObjectURL(blob);
+  const link = document.createElement("a");
+  link.href = url;
+  link.download = "downloaded-image.jpg";
+
+  // Trigger download
+  document.body.appendChild(link);
+  link.click();
+  document.body.removeChild(link);
+
+  // Clean up
+  URL.revokeObjectURL(url);
+};
+
+export const drawShape = (
+  shape: string,
+  startX: number,
+  startY: number,
+  x: number,
+  y: number,
+  id: string
+): void => {
+  const canvas: HTMLElement | null = document.getElementById(id);
+  if (canvas) {
```

Comment:
just for better look
```suggestion
  if (!canvas) return;
```